### ISSUE TITLE
New arguments for the CommandAPI

### ISSUE THREAD

**Issue by JorelAli** [JorelAli/1.13-Command-API#32]
**Please describe your suggestion**
Add new argument options to the CommandAPI. In particular:
* `ScoreboardTeamArgument` - Lets you access Bukkit's [Team](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/scoreboard/Team.html) class
* `ScoreboardObjectiveArgument` - Lets you access Bukkit's [Objective](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/scoreboard/Objective.html) class
* `RecipeArgument` - Lets you access Bukkit's [Recipe](https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/inventory/Recipe.html) class

**Please supply examples of the desired inputs and outputs**
_Team Example Usage:_
```java
(sender, args) -> {
    Team team = (Team) args[0];
    team.addPlayer((Player) args[1]);
    //other team related code
}
```
_Objective Example Usage:_
```java
(sender, args) -> {
    Objective obj = (Objective) args[0];
    obj.setDisplayName((String) args[1]);
    //other objective related code
}
```
_Recipe Example Usage:_
```java
(sender, args) -> {
    Recipe r = (Recipe) args[0];
    if(r.getResult().isSimilar​(/* the player's hand item? */)) {
        //code
    }
}
```

**Comment by HielkePlugins** [comment]
Looks really good! 
But doesn't vanilla use dynamic string arguments as scoreboard/team arguments? 
Like with `/scoreboard remove <name>` you can type anything you want, not just the scoreboard names.

I got scoreboard arguments to work just fine with this:

```java
    private DynamicSuggestedStringArgument objectives = new DynamicSuggestedStringArgument(() ->
    {
        List<String> names = new ArrayList<>();
        Set<Objective> objectiveSet = Bukkit.getScoreboardManager().getMainScoreboard().getObjectives();
        for(Objective objective: objectiveSet){
            names.add(objective.getName());
        }
        return names.toArray(new String[0]);
    });
```
Would your scoreboard/team arguments work the same way? Or will it only accept scoreboard/team names?

**Comment by JorelAli** [comment]
> Looks really good!
> But doesn't vanilla use dynamic string arguments as scoreboard/team arguments?
> Like with `/scoreboard remove <name>` you can type anything you want, not just the scoreboard names.
> 
> I got scoreboard arguments to work just fine with this:
> 
> ```java
>     private DynamicSuggestedStringArgument objectives = new DynamicSuggestedStringArgument(() ->
>     {
>         List<String> names = new ArrayList<>();
>         Set<Objective> objectiveSet = Bukkit.getScoreboardManager().getMainScoreboard().getObjectives();
>         for(Objective objective: objectiveSet){
>             names.add(objective.getName());
>         }
>         return names.toArray(new String[0]);
>     });
> ```
> 
> Would your scoreboard/team arguments work the same way? Or will it only accept scoreboard/team names?

It has taken me 2 hours to finally inform you that you are indeed right and that there's basically no point to me creating objectives or teams because vanilla basically uses a dynamic suggested string. Due to this, I have chosen not to implement the things outlined in this issue. However, I may create some templates for Dynamic Suggested Arguments which are prepopulated with various things which may be useful.

**Comment by JorelAli** [comment]
For completeness, this has been implemented in versions 3.0 and 2.1.
